### PR TITLE
Adds the advanced health analyzer

### PR DESCRIPTION
**Pull request by mistyLuminescence**
Adds the advanced health analyzer!

Like a regular health analyzer, but:

- Gives specific details on broken bone and internal bleeding locations

- Notices low (1-10) brain damage

- Gives less vague radiation readings (minor/moderate/severe/critical)

- Gives specific details on reagents in blood and stomach

Note: this doesn't make the body scanner obsolete - we still need that to detect shrapnel, implants, organ damage, genetic damage quantity and paralysis time.

Currently, it's included in the traitor surgery kit (45 TC), the merc mediborg loadout, and there's a module for it in Robotics (requires diamond) that can be applied to borgs (like the jetpack module) - surgeon borgs especially will like this. I'd love to see it in R&D or the CMO loadout sometime too.

For all you research nerds, it can be deconstructed to push biotech a liiittle higher than the current cap.

### CODE REVIEW

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -100,39 +106,75 @@ REAGENT SCANNER
...
+		if(advscan == 1)
+			if(M.radiation >= 75)
+				user.show_message("<span class='warning'>Critical levels of radiation detected. Immediate treatment advised.</span>")
+			else if(M.radiation >= 50)
+				user.show_message("<span class='warning'>Severe levels of radiation detected.</span>")
+			else if(M.radiation >= 25)
+				user.show_message("<span class='warning'>Moderate levels of radiation detected.</span>")
+			else if(M.radiation >= 1)
+				if(prob(5))
+					user.show_message("Low levels of radiation detected. Should've worn a radiation suit.</span>")
```
Comment (Anewbe) [inline code/game/objects/items/devices/scanners.dm]: We don't need this message.

File: code/game/objects/items/devices/scanners.dm [hunk 2]
```diff
@@ -100,39 +106,75 @@ REAGENT SCANNER
...
 			var/unknown = 0
 			var/reagentdata[0]
+			var/unknownreagents[0]
 			for(var/A in C.reagents.reagent_list)
 				var/datum/reagent/R = A
 				if(R.scannable)
 					reagentdata["[R.id]"] = "<span class='notice'>    [round(C.reagents.get_reagent_amount(R.id), 1)]u [R.name]</span>"
 				else
 					unknown++
+					unknownreagents["[R.id]"] = "<span class='notice'>    [round(C.reagents.get_reagent_amount(R.id), 1)]u [R.name]</span>"
```
Comment (Anewbe) [inline code/game/objects/items/devices/scanners.dm]: I'm not really a fan of this change.

File: code/game/objects/items/devices/scanners.dm [hunk 3]
```diff
@@ -145,6 +187,11 @@ REAGENT SCANNER
 		user.show_message("<span class='warning'>Severe brain damage detected. Subject likely to have a traumatic brain injury.</span>")
 	else if (M.getBrainLoss() >= 10)
 		user.show_message("<span class='warning'>Significant brain damage detected. Subject may have had a concussion.</span>")
+	else if (M.getBrainLoss() >= 1 && advscan == 1)
+		if(prob(5))
+			user.show_message("<span class='warning'>Minor brain damage detected. Professional analysis: not the sharpest tool in the shed.</span>")
```
Comment (Anewbe) [inline code/game/objects/items/devices/scanners.dm]: We don't need this message.

**Comment by mistyLuminescence** [comment]
Snark removed, Travis (hopefully) fixed
fight me travis

**Comment by mistyLuminescence** [comment]
PR updated with tiers. 

Tier 0: regular health analyzer.
Tier 1 (improved): IB, bone fractures, beneficial stomach reagent quantity. 
Tier 2 (advanced): the above, plus radiation severity and minor brain damage detection.
Tier 3 (refined): the above, plus identity and quantity of unknown stomach reagents.

Tier 0 is the health analyzer we all know and love.
Tier 1 is adminspawn only, but I'd like to see it in R&D soon.
Mercenary medical borgs, the antag surgery kit, and stationbound medborgs (through an upgrade in Robotics) get tier 2.
Tier 3 is adminspawn only, and will probably remain this way.

**Pull request merged**